staging/src/k8s.io/apiserver/pkg/endpoints/discovery/aggregated/metrics.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aggregated
+
+import (
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+var (
+	regenerationCounter = metrics.NewCounter(
+		&metrics.CounterOpts{
+			Name:           "aggregator_discovery_aggregation_count_total",
+			Help:           "Counter of number of times discovery was aggregated",
+			StabilityLevel: metrics.ALPHA,
+		},
+	)
+	regenerationDurationGauge = metrics.NewGauge(
```

Comment:
Do we need both? Doesn't the gauge also count the number of occurences? I think we discussed that, sorry if I keep forgetting 😅

Reply:
Yes both are needed. https://pkg.go.dev/github.com/prometheus/client_golang/prometheus?utm_source=godoc#Gauge, I was planning to use this to record only the latest aggregation duration.

Reply:
Gauge seems like the wrong choice here? Aren't we trying to gather duration that we'll want to get averages/median times?

Reply:
In a sense. Gathering durations for something like api requests make sense because we have multiple request for the same discovery doc so they can be bucketed.

For reaggregation, an unchanged discovery doc will only have one data point. Realistically reaggregation only happen when CRDs are modified and the time should be proportional to the # custom resources. I think we really only care about the latest time after all the CRDs are installed (final state). Taking the average doesn't make as much sense because we don't really care about the duration for partial states (eg: half the CRDs are applied).

(I'm using CRD/aggregated apiservers synonymously but the same logic applies to both)

Reply:
Discussed offline, dropping the duration metric since the number is negligible and we will have pretty good visibility with the renegeneration counter as well as the request duration instrumentation.